server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -205,6 +235,77 @@ public void onHit(Key key, CacheValue<BytesReference> cacheValue) {
     @Override
     public void onCached(Key key, BytesReference value, TierType tierType) {
         key.entity.onCached(key, value, tierType);
+        updateDiskCleanupKeyToCountMap(new CleanupKey(key.entity, key.readerCacheKeyId), tierType);
```

Comment:
yep, i thought the cleanupKey would be more appropriate here since we are in the context of updating the map with fields of cleanupKey. 
But recreating cleanupKey only for this seems unnecessary like u said.